### PR TITLE
ledger refactoring: remove assets limits

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Remove the existing asset / app limits.

## Test Plan

Use existing tests.

### CODE REVIEW

File: netdeploy/remote/deployedNetwork.go [hunk 1]
```diff
@@ -587,6 +587,12 @@ func accountsNeeded(appsCount uint64, assetCount uint64, params config.Consensus
...
 
 	maxApps = uint64(params.MaxAppsCreated)
+	// TODO : given that we've added unlimited app support, we should revise this
+	// code so that we'll have control on how many app/account we want to create.
+	// for now, I'm going to keep the previous max values until we have refactored this code.
+	if maxApps == 0 {
+		maxApps = 10
+	}
 
 	if maxApps > 0 {
```
Comment (algonautshant) [inline netdeploy/remote/deployedNetwork.go:597]: This condition is alway true, since `maxApps` can no longer be 0.

**Review by algonautshant — APPROVED** [review]
Looks great!
Couple of minor comments.

File: ledger/apply/application.go [hunk 2]
```diff
@@ -70,11 +70,16 @@ func createApplication(ac *transactions.ApplicationCallTxnFields, balances Balan
 
 	// Make sure the creator isn't already at the app creation max
 	maxAppsCreated := balances.ConsensusParams().MaxAppsCreated
-	if totalAppParams >= uint32(maxAppsCreated) {
+	if maxAppsCreated > 0 && totalAppParams >= uint32(maxAppsCreated) {
 		err = fmt.Errorf("cannot create app for %s: max created apps per acct is %d", creator.String(), maxAppsCreated)
 		return
 	}
 
+	if !balances.ConsensusParams().Application {
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/apply/application.go:78]: sounds reasonable.

File: netdeploy/remote/deployedNetwork.go (hunk 1 above)
Comment (tsachiherman) [reply netdeploy/remote/deployedNetwork.go:597]: true; but I didn't want to change this logic at this time. I think that we will need more changes around this tool to align with the unlimited assets support, and I've just retained backward compatibility for now.

File: ledger/apply/application.go [hunk 3]
```diff
@@ -237,9 +242,12 @@ func optInApplication(balances Balances, sender basics.Address, appIdx basics.Ap
 
 	// Make sure the user isn't already at the app opt-in max
 	maxAppsOptedIn := balances.ConsensusParams().MaxAppsOptedIn
-	if totalAppLocalState >= uint64(maxAppsOptedIn) {
+	if maxAppsOptedIn > 0 && totalAppLocalState >= uint64(maxAppsOptedIn) {
 		return fmt.Errorf("cannot opt in app %d for %s: max opted-in apps per acct is %d", appIdx, sender.String(), maxAppsOptedIn)
 	}
+	if !balances.ConsensusParams().Application {
+		return fmt.Errorf("cannot opt in app %d for %s: applications not supported", appIdx, sender.String())
+	}
```
Comment (algonautshant) [inline ledger/apply/application.go:250]: Ooops! the same issue here. Maybe this can go before line 246.

**Review by algonautshant — DISMISSED** [review]

**Review by algonautshant — DISMISSED** [review]

File: ledger/catchpointwriter_test.go [hunk 4]
```diff
@@ -69,8 +69,12 @@ func makeTestEncodedBalanceRecordV5(t *testing.T) encodedBalanceRecordV5 {
 		AuthAddr:           basics.Address(crypto.Hash([]byte{1, 2, 3, 4})),
 	}
 	currentConsensusParams := config.Consensus[protocol.ConsensusCurrentVersion]
-
-	for assetCreatorAssets := 0; assetCreatorAssets < currentConsensusParams.MaxAssetsPerAccount; assetCreatorAssets++ {
+	maxAssetsPerAccount := currentConsensusParams.MaxAssetsPerAccount
+	// if the number of supported assets is unlimited, create only 1000 for the purpose of this unit test.
+	if maxAssetsPerAccount == 0 {
+		maxAssetsPerAccount = 1000
```
Comment (algorandskiy) [inline ledger/catchpointwriter_test.go:75]: same as before, use a real value from a prev version

File: ledger/catchpointwriter_test.go [hunk 5]
```diff
@@ -97,6 +101,12 @@ func makeTestEncodedBalanceRecordV5(t *testing.T) encodedBalanceRecordV5 {
 
 	maxApps := currentConsensusParams.MaxAppsCreated
 	maxOptIns := currentConsensusParams.MaxAppsOptedIn
+	if maxApps == 0 {
+		maxApps = 10
```
Comment (algorandskiy) [inline ledger/catchpointwriter_test.go:105]: and here

File: ledger/evalbench_test.go [hunk 6]
```diff
@@ -109,6 +109,9 @@ type BenchAppOptInsTxnGenerator struct {
 func (g *BenchAppOptInsTxnGenerator) Prepare(tb testing.TB, addrs []basics.Address, keys []*crypto.SignatureSecrets, rnd basics.Round, gh crypto.Digest) ([]transactions.SignedTxn, int) {
 	maxLocalSchemaEntries := config.Consensus[g.Proto].MaxLocalSchemaEntries
 	maxAppsOptedIn := config.Consensus[g.Proto].MaxAppsOptedIn
+	if maxAppsOptedIn == 0 {
+		maxAppsOptedIn = 10
```
Comment (algorandskiy) [inline ledger/evalbench_test.go:113]: and here

File: netdeploy/remote/deployedNetwork.go [hunk 7]
```diff
@@ -587,6 +587,12 @@ func accountsNeeded(appsCount uint64, assetCount uint64, params config.Consensus
 	var nAppAcct uint64
 
 	maxApps = uint64(params.MaxAppsCreated)
+	// TODO : given that we've added unlimited app support, we should revise this
+	// code so that we'll have control on how many app/account we want to create.
+	// for now, I'm going to keep the previous max values until we have refactored this code.
+	if maxApps == 0 {
+		maxApps = 10
```
Comment (algorandskiy) [inline netdeploy/remote/deployedNetwork.go:594]: config.Consensus[protocol.ConsensusV30]. MaxAppsCreated

File: netdeploy/remote/deployedNetwork.go [hunk 8]
```diff
@@ -598,6 +604,12 @@ func accountsNeeded(appsCount uint64, assetCount uint64, params config.Consensus
 	var maxAssets uint64
 	var nAssetAcct uint64
 	maxAssets = uint64(params.MaxAssetsPerAccount)
+	// TODO : given that we've added unlimited asset support, we should revise this
+	// code so that we'll have control on how many asset/account we want to create.
+	// for now, I'm going to keep the previous max values until we have refactored this code.
+	if maxAssets == 0 {
+		maxAssets = 1000
```
Comment (algorandskiy) [inline netdeploy/remote/deployedNetwork.go:611]: same here and below

**Review by algorandskiy — COMMENTED** [review]
Looks good, but please change from hard-coded numeric literals (10, 1000) to actual values from V30

**Review by algorandskiy — DISMISSED** [review]

File: data/basics/userBalance_test.go [hunk 9]
```diff
@@ -190,8 +190,11 @@ func TestEncodedAccountDataSize(t *testing.T) {
 		}
 		maxLocalState[maxKey] = maxValue
 	}
-
-	for appCreatorApps := 0; appCreatorApps < currentConsensusParams.MaxAppsCreated; appCreatorApps++ {
+	maxAppsCreate := currentConsensusParams.MaxAppsCreated
+	if maxAppsCreate == 0 {
+		maxAppsCreate = config.Consensus[protocol.ConsensusV30].MaxAppsCreated
```
Comment (cce) [inline data/basics/userBalance_test.go:195]: wouldn't this revert an unlimited MaxAppsCreated to 10?

Comment (cce) [reply data/basics/userBalance_test.go:195]: Oh I see you are using this to generate test data, so just make 10 apps if apps are unlimited.

Comment (tsachiherman) [reply data/basics/userBalance_test.go:195]: exactly.

**Pull request merged**